src/pudl/transform/eia861.py

```diff
@@ -1288,6 +1252,148 @@ def distributed_generation(tfr_dfs):
+    df_pre_2010_tech = raw_dg_tech[raw_dg_tech['report_date'] < '2010-01-01']
+    df_post_2010_tech = raw_dg_tech[raw_dg_tech['report_date'] >= '2010-01-01']
+    df_pre_2010_misc = raw_dg_misc[raw_dg_misc['report_date'] < '2010-01-01']
+    df_post_2010_misc = raw_dg_misc[raw_dg_misc['report_date'] >= '2010-01-01']
+
+    logger.info(
+        'Converting pct values into mw values for distributed generation misc table')
+    transformed_dg_misc = (
+        df_pre_2010_misc.assign(
+            distributed_generation_owned_capacity_mw=lambda x: _pct_to_mw(
+                x, 'distributed_generation_owned_capacity_pct'),
+            backup_capacity_mw=lambda x: _pct_to_mw(x, 'backup_capacity_pct'),
+        ).append(df_post_2010_misc)
+        .drop(['distributed_generation_owned_capacity_pct',
+               'backup_capacity_pct',
+               'total_capacity_mw'], axis=1)
+    )
+
+    logger.info(
+        'Converting pct values into mw values for distributed generation misc table')
+    transformed_dg_tech = (
+        df_pre_2010_tech.assign(
+            combustion_turbine_capacity_mw=lambda x: (
+                _pct_to_mw(x, 'combustion_turbine_capacity_pct')),
+            hydro_capacity_mw=lambda x: _pct_to_mw(x, 'hydro_capacity_pct'),
+            internal_combustion_capacity_mw=lambda x: (
+                _pct_to_mw(x, 'internal_combustion_capacity_pct')),
+            other_capacity_mw=lambda x: _pct_to_mw(x, 'other_capacity_pct'),
+            steam_capacity_mw=lambda x: _pct_to_mw(x, 'steam_capacity_pct'),
+            wind_capacity_mw=lambda x: _pct_to_mw(x, 'wind_capacity_pct'),
+        ).append(df_post_2010_tech)
+        .drop([
+            'combustion_turbine_capacity_pct',
+            'hydro_capacity_pct',
+            'internal_combustion_capacity_pct',
+            'other_capacity_pct',
+            'steam_capacity_pct',
+            'wind_capacity_pct',
+            'total_capacity_mw'], axis=1
+        )
+    )
+
+    ###########################################################################
+    # Tidy Data
+    ###########################################################################
+
+    logger.info('Tidying Distributed Generation Tech Table')
+    tidy_dg_tech, tech_idx_cols = _tidy_class_dfs(
+        df=transformed_dg_tech,
+        df_name='Distributed Generation Tech Component Capacity',
+        idx_cols=idx_cols,
+        class_list=pc.TECH_CLASSES,
+        class_type='tech_class',
+    )
+
+    logger.info('Tidying Distributed Generation Fuel Table')
+    tidy_dg_fuel, fuel_idx_cols = _tidy_class_dfs(
+        df=raw_dg_fuel,
+        df_name='Distributed Generation Fuel Percent',
+        idx_cols=idx_cols,
+        class_list=pc.FUEL_CLASSES,
+        class_type='fuel_class',
+    )
+
+    tfr_dfs["distributed_generation_tech_eia861"] = tidy_dg_tech
+    tfr_dfs["distributed_generation_fuel_eia861"] = tidy_dg_fuel
+    tfr_dfs["distributed_generation_misc_eia861"] = transformed_dg_misc
```

Comment:
do we need to delete the original table from `tfr_dfs` as well??

Reply:
Yes! you're right